web/src/compose_recipient.js

```diff
@@ -198,32 +202,27 @@ export function update_compose_for_message_type(message_type, opts) {
 }
 
 export function on_compose_select_recipient_update() {
-    let message_type = "stream";
+    const prev_message_type = compose_state.get_message_type();
+
+    let curr_message_type = "stream";
     if (selected_recipient_id === DIRECT_MESSAGE_ID) {
-        message_type = "private";
+        curr_message_type = "private";
     }
-    compose_state.set_message_type(message_type);
-    if (message_type === "private") {
-        // TODO: In theory, we could do something more lightweight in
-        // the case it's already that value, but doing nothing would
-        // display the wrong and fail to update focus properly.
-        switch_message_type("private");
 
-        if (compose_state.private_message_recipient().length === 0) {
-            $("#private_message_recipient").trigger("focus").trigger("select");
-        }
-    } else {
+    if (prev_message_type !== curr_message_type) {
+        switch_message_type(curr_message_type);
+    }
```

Comment:
I'm pretty sure that we were doing `switch_message_type` unconditionally before for a reason, related to some side effect that had -- maybe it was this focus behavior you're adjusting here; do you remember?

Reply:
Yes, that seems to be case.